sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -197,10 +215,17 @@ def fit(self, X, y=None):
         X = check_array(X, dtype=[X_DTYPE], ensure_all_finite=False)
         max_bins = self.n_bins - 1
 
+        subsampling_probabilities = None
+        if sample_weight is not None:
+            subsampling_probabilities = sample_weight / np.sum(sample_weight)
+
         rng = check_random_state(self.random_state)
         if self.subsample is not None and X.shape[0] > self.subsample:
-            subset = rng.choice(X.shape[0], self.subsample, replace=False)
+            subset = rng.choice(
+                X.shape[0], self.subsample, p=subsampling_probabilities, replace=False
```

Comment:
I think we have to pass `replace=True` to get the statistical test to pass.

```suggestion
                X.shape[0], self.subsample, p=subsampling_probabilities, replace=True
```

Reply:
I tried locally and I confirm that this makes all the statistical tests of https://github.com/scikit-learn/scikit-learn/pull/29641#issuecomment-2743372458 pass.